### PR TITLE
TF-1194 [Part II] Apply new identity design to edit identity screen

### PR DESCRIPTION
**Pull request by sherlockvn**
https://user-images.githubusercontent.com/43041967/212958965-0dab45a0-9c14-4a4c-847f-79864fb50970.mp4

mobile: 

https://user-images.githubusercontent.com/43041967/213262942-ad0de76f-15cf-43fd-bc2f-693fe4d95fe2.mp4

### CODE REVIEW

File: lib/features/manage_account/domain/usecases/edit_default_identity_interactor.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    try {
+      yield Right(EditDefaultIdentityLoading());
+
+      final listDefaultIdentities = await _getDefaultIdentities(accountId);
+
+      // if the identity need to update in listDefaultIdentities, remove it
+      listDefaultIdentities
+          ?.removeWhere(
+              (identity) => identity.id == editIdentityRequest.identityId
+          );
```
Comment (hoangdat) [inline lib/features/manage_account/domain/usecases/edit_default_identity_interactor.dart:36]: how to make it clear and short? one tip: create a method. Ex:
```suggestion
      listDefaultIdentities
          ?.removeWhere((identity) => requestIdentityStillDefault(editIdentityRequest.identityId, identityId));
          
          
       .......
       bool requestIdentityStillDefault(requestIdentity, defaultIdentityId) {
          //bla bla bla
       }
```

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 2]
```diff
@@ -243,6 +247,19 @@ class IdentityCreatorController extends BaseController {
     }
   }
 
+  void _setUpDefaultIdentity(List<Identity>? identities) {
+    final listDefaultIdentityIds = _identityUtils
+        .getSmallestOrderedIdentity(identities)
+        ?.map((identity) => identity.id!);
+    
+    // if all identities have same sortOrder, it means it don't have default identity
```
Comment (hoangdat) [inline lib/features/identity_creator/presentation/identity_creator_controller.dart:255]: avoid using comment by using method name

File: model/pubspec.lock [hunk 3]
```diff
@@ -420,7 +420,7 @@ packages:
     description:
       path: "."
       ref: master
-      resolved-ref: ea636193b15a49e9d59dab143e957f529e1bd836
+      resolved-ref: "1df4203f3c7cb5f24ee68f3e1060efb7283dd3fd"
```
Comment (dab246) [inline model/pubspec.lock:423]: What did we update on `jmap_dart_client`?

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 4]
```diff
@@ -164,7 +167,7 @@ class IdentityCreatorController extends BaseController {
     if (accountId != null) {
       consumeState(_getAllIdentitiesInteractor.execute(
         accountId!,
-        properties: Properties({'email'})
+        properties: Properties({'email', 'sortOrder'})
```
Comment (hoangdat) [inline lib/features/identity_creator/presentation/identity_creator_controller.dart:170]: What happens if we request this property `sortOrder` with the JMAP server not support default identity?

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — APPROVED** [review]
lgtm

**Pull request merged**